migrations/migration.py

```diff
@@ -0,0 +1,231 @@
+"""Migration Module
+
+Captures common methods needed by the system for migration
+"""
+
+import os
+from typing import List
+from datetime import datetime
+import uuid
+import imp
+
+from migrations.migration_resource import MigrationManager
+from migrations.utils import Node, LinkedList
+from isacc_messaging.audit import audit_entry
+from isacc_messaging.models.fhir import (
+    first_in_bundle,
+)
+
+
+class Migration:
+    def __init__(self, migrations_dir=None):
+        if migrations_dir is None:
+            migrations_dir = os.path.join(os.path.dirname(__file__), "versions")
+        self.migrations_dir = migrations_dir
+        self.migration_sequence = LinkedList()
+        self.build_migration_sequence()
+
+    def build_migration_sequence(self):
+        migration_files: list = self.get_migration_files()
+        migration_nodes: dict = {}
+
+        for filename in migration_files:
+            migration: str = filename[:-3]
+            migration_nodes[migration] = self.get_previous_migration_id(migration)
+
+        if len(migration_files) > 0:
+            try:
+                self.migration_sequence.build_list_from_dictionary(migration_nodes)
+            except:
+                error_message = "Cycle detected in migration sequence"
+                audit_entry(error_message, level='error')
+                raise ValueError(error_message)
+        else:
```

Comment:
I don't believe you want to generate an error here.  We should be able to add the migration library to a system, without requiring any migration scripts, right?

Reply:
never mind - just the variable name, not an error being raised.